apps/prairielearn/src/batched-migrations/20250930221841_json_points_fields__backfill.ts

```diff
@@ -0,0 +1,82 @@
+import { z } from 'zod';
+
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import * as namedLocks from '@prairielearn/named-locks';
+import { queryRow, queryRows } from '@prairielearn/postgres';
+
+import { type Course, CourseSchema } from '../lib/db-types.js';
+import { createServerJob } from '../lib/server-jobs.js';
+import { getLockNameForCoursePath } from '../models/course.js';
+import { syncDiskToSqlWithLock } from '../sync/syncFromDisk.js';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    const result = await queryRow(
+      'SELECT MAX(id) as max from pl_courses;',
+      z.coerce.bigint().nullable(),
+    );
+    return {
+      min: 1n,
+      max: result,
+      batchSize: 10,
+    };
+  },
+
+  async execute(min: bigint, max: bigint): Promise<void> {
+    const courses = await queryRows(
+      'SELECT * FROM pl_courses WHERE id >= $min AND id <= $max AND deleted_at IS NULL',
+      { min, max },
+      CourseSchema,
+    );
+
+    const errors: Error[] = [];
+    for (const course of courses) {
+      try {
+        await syncCourse(course);
+      } catch (err) {
+        errors.push(
+          new Error(`Failed to sync course ${course.id} (${course.short_name}): ${err}`, {
```

Comment:
Nit, error details will already be included by `cause` so it shouldn't be necessary to repeat the error message here. You can test that this logs reasonable things by adding an artificial error to `syncCourse` and checking the batched migration error logs.

```suggestion
          new Error(`Failed to sync course ${course.id} (${course.short_name})`, {
```